### ISSUE TITLE
phpbbSEO destroys ajax on switch topic page and mChat reload function

### ISSUE THREAD

**Issue by Sarr** [phpBBSEO/usu#101]
Hello. After adding pre-generated .htaccess phpbbSEO works fine except it destroys ajax on switch topic page and mChat reload function. mChat refreshes without end, and changing topic page while logged in generates ajax error, and it won't switch page.

I have latest versions of everything, plus nodupe and meta mod of yours installed and configured.
Ask me for anything and I'll help. My forums are here: http://forum.swordcoast.pl/

**Comment by Galixte** [comment]
This extension has not been updated since the month of november 2014, are you going use a non updated extension for a long time ?

**Comment by Sarr** [comment]
Well, they say it's still in the works and should work fine. It's working great in fact on latest phpbb 3.1.6, there's only this ajax problem, so I thought it's worth fixing. This is the best SEO extension for phpbb, period. No other matches it. And there's only tas2580 SEO URLs for phpbb 3.1 from here: https://tas2580.net/downloads/download-11.html

Also see: http://www.phpbb-seo.com/en/mixed-seo-url/article10130.html
http://www.phpbb-seo.com/en/site/article9580-60.html

**Comment by cYbercOsmOnauT** [comment]
It's wrong to say "there's only this ajax problem". There are more. I changed some places for a customer of mine.

For example if a "too high" page is surfed directly phpbbSEO redirects to the highest possible page with a 301. But the browsers are caching this "permanent" redirection internally and users get problem to see the newest page if there are more posts. You have to use 302 for things like this.

**Comment by Galixte** [comment]
For me, this extension is useless until a new developper will take his time to continue.

Otherwise, the best SEO tool is your “original” board content, see this: http://phpbb.hifikabin.me.uk/app.php/page/seo.

**Comment by cYbercOsmOnauT** [comment]
@Galixte Nice written from HiFiKabin for newly starting sites. But there are people out there who don't want to lose their already traced links on google. I am not talking about _normal sites_ but about sites who are and want to stay on page 1 position 1-5 of the results page.

As for the points HiFiKabin wrote in his post.
1. **Keywords** - Post is 6 years old and only half right. You maybe don't need meta keywords anymore but the content of the page should be written _around_ the keywords.
2. **Static URLs** - Post is even 7 years old. Yes Google can differenciate dynamic urls but you cannot be sure that they know which url parameters are unneeded. You end up with _double content_ which gives you alot _minus points_ for the ranking of your site.
3. **HTTPS** - Last year they said that it has _less weight_. And even inside the post they say that they may decide to strengthen this.. and wow.. they did. Sites that use https get better ranking already.

Not SEO but also important for ranking is beeing _mobile friendly_. Google gives alot ranking points for sites which are responsive.

**Comment by neuropass** [comment]
I agree with @cYbercOsmOnauT We just need some polishing on this extension. Also regarding the last point for the mobile friendly website. 

It really is part of the SEO as well. You c an use  premium template to increase your SEO. See [SiteSplat's themes](http://themeforest.net/user/themesplat/portfolio?ref=ThemeSplat) - I'm currently using FLATBOOTS.  My forum ranks much better now (even though we are not a busy community anyway as we are old school OS modders ), its faster than any other themes I tried and my user retention increased as well as they seem to be happy with the options and usability. Good luck!

**Comment by Sarr** [comment]
@Galixte I have 3 gaming portals that are top 1-5 in search results in my whole country. Plus some other google page 1 sites. So I know that yes, urls have some impact. Content is and always was, and always should be a major factor. But I am sure that if you have a site with good content and strange urls, and a competitor with very user-friendly (yes, SEO urls are more user-friendly), then Google will rank the site with SEO urls higher.
Plus of course you should exclude some addresses from google indexing, make site responsive, fast loading, secure, and many other things are to be done.

@cYbercOsmOnauT well, not really wrong to say, that was my opinion from my observations, in my case. It's what I discovered through my usage. I wouldn't be even so sure that you've found all the bugs and problems yet.

But if you fixed those issues, you can upload this build to github, right? Your customer paid you for your work and it's all fine, but the software is free. The community would benefit.

@neuropass you are just spamming here, advertising for a paid product.

**Comment by cYbercOsmOnauT** [comment]
@Sarr my fixes are mostly fastly patched things for my customer which I don't want to release to the wild. Some stuff is even _hardcoded_.

But yes. I could diff my version with the actual one here and see if I can give some of my changes to the community. My customer allows me to do so. As you can see in my repositories there are some stuff I coded for money and was allowed to release for everyone.
